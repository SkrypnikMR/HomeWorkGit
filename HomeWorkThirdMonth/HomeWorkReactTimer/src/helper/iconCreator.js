import React from 'react';

export const icon = (mode) => (mode === 'light' ? <>&#9728;</> : <>&#127770;</>);
