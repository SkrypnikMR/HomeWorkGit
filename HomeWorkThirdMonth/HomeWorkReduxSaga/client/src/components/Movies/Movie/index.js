import { connect } from 'react-redux';
import Movie from './Movie.jsx';
import { deleteMovie, updateMovie } from '../../../store/movies/actions';
import { getMovieById, getIsLoading } from '../../../store/movies/selectors';

const mapStateToProps = (state, props) => ({
    movie: getMovieById(state, props),
    isLoading: getIsLoading(state)
});

const mapDispatchToProps = { deleteMovie, updateMovie };

export default connect(mapStateToProps, mapDispatchToProps)(Movie);
