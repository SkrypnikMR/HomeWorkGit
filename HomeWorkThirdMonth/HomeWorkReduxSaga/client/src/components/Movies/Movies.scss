.movies{
    top: 70px;
    width: 100%;
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    align-items: center;
    margin-top: 30px;
}
