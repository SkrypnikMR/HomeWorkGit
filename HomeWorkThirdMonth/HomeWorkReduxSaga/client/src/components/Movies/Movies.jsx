import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Movie from './Movie';
import Loader from '../Loader';
import './Movies.scss';

class Movies extends Component {
    componentDidMount() {
        const { getMovies } = this.props;
        getMovies();
    }

    render() {
        const { movies, isLoading } = this.props;

        return (
            <div className='movies'>
                {isLoading
                    ? <Loader />
                    : null}
                {movies.map((movie) => (<Movie id={movie.id} key={movie.id} />))}
            </div>
        );
    }
}

Movies.propTypes = {
    movies: PropTypes.arrayOf(PropTypes.object),
    getMovies: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
};

export default Movies;
