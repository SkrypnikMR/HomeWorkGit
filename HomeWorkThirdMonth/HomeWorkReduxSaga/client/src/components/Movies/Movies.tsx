import React, { useEffect } from 'react';
import Movie from './Movie';
import Loader from '../Loader';
import './Movies.scss';

interface IMovies {
    getMovies: () => void;
    movies: { id: string; }[];
    isLoading: boolean;
}

const Movies = ({ getMovies, movies, isLoading }: IMovies) => {
    useEffect(() => { getMovies() }, []);
    return (
        <div className='movies'>
            {isLoading
                ? <Loader />
                : null}
            {movies.map((movie) => (<Movie id={movie.id} key={movie.id} />))}
        </div>
    );
}


export default Movies;
