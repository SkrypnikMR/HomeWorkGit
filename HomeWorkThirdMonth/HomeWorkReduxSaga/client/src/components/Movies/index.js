import { connect } from 'react-redux';
import Movies from './Movies.jsx';
import { getMovies } from '../../store/movies/asyncActions.js';
import { getMoviesStoreList, getIsLoading } from '../../store/movies/selectors';

const mapStateToProps = (state) => ({
    movies: getMoviesStoreList(state),
    isLoading: getIsLoading(state),
});

const mapDispatchToProps = { getMovies };

export default connect(mapStateToProps, mapDispatchToProps)(Movies);
