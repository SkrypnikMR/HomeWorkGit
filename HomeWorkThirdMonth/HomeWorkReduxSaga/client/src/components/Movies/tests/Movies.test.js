import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../__tests__/helper';
import Movies from '../Movies';

const mockStore = configureStore();
const store = mockStore({
    movies: [{ id: 1, cover: 'cover', title: 'title', description: 'desc' },
    { id: 2, cover: 'cover', title: 'title', description: 'desc' }],
    moviesIds: [],
    isLoading: false,
    newMovie: {},
});

describe('Movies', () => {
    const getMovies = jest.fn()
    it('Should match snapshot', () => {
        const component = shallowSmart(<Movies
            isLoading={false}
            getMovies={getMovies}
            movies={[]}
        />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should have Loader', () => {
        const component = mountSmart(<Movies
            isLoading={true}
            getMovies={getMovies}
            movies={[]}
        />, store);
        expect(component.find('Loader')).toHaveLength(1);
    });
    it('Should have Movies', () => {
        const component = mountSmart(<Movies
            isLoading={true}
            getMovies={getMovies}
            movies={[{ id: 1 }, { id: 2 }]}
        />, store);
        expect(component.find('Movie')).toHaveLength(2);
    });
});
