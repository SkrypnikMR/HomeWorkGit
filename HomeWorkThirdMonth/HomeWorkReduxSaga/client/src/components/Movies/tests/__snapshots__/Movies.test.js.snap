// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Should match snapshot 1`] = `"<div class=\\"movies\\"></div>"`;
